src/mlpack/methods/reinforcement_learning/environment/pendulum.hpp

```diff
@@ -140,6 +155,15 @@ class Pendulum
         -maxAngularVelocity), maxAngularVelocity);
     nextState.Theta() = theta + newAngularVelocity * dt;
 
+    // Check if the episode has terminated
+    bool done = IsTerminal(nextState);
+
+    // Do not reward the agent if time ran out.
+    if (done && maxSteps != 0 && timeStepsPerformed >= maxSteps)
```

Comment:
Since we are going with ``maxSteps`` I think it makes sense to call this ``stepsPerformed``?